lib/ansible/modules/cloud/amazon/neptune_cluster.py

```diff
@@ -0,0 +1,356 @@
+  backup_window:
+    description:
+    - The daily time range during which automated backups are created if automated backups are enabled using
+      the `retention_period` parameter.
+  maintenance_window:
+    description:
+    - The weekly time range during which system maintenance can occur, in Universal Coordinated Time (UTC).
+  replication_source:
+    description:
+    - The Amazon Resource Name (ARN) of the source DB instance or DB cluster if this DB cluster is created as a Read Replica.
+  encrypted:
+    description:
+    - Specifies whether the DB cluster is encrypted.
+    type: bool
+    default: false
+  kms_key_id:
+    description:
+    - The AWS KMS key identifier for an encrypted DB cluster.
+  pre_signed_url:
+    description:
+    - A URL that contains a Signature Version 4 signed request for the CreateDBCluster action to be called in the source AWS Region
+      where the DB cluster is replicated from.
+    - You only need to specify PreSignedUrl when you are performing cross-region replication from an encrypted DB cluster.
+  enable_iam_auth:
+    description:
+    - True to enable mapping of AWS Identity and Access Management (IAM) accounts to database accounts, and otherwise false.
+    type: bool
+    default: false
+  skip_final_snapshot:
+    description:
+    - Determines whether a final DB cluster snapshot is created before the DB cluster is deleted.
+    type: bool
+    default: false
+  final_snapshot_id:
+    description:
+    - The DB cluster snapshot identifier of the new DB cluster snapshot created when `skip_final_snapshot` is set to false .
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = r'''
+  - name: Create a new graph database custer
+    neptune_cluster:
+      name: "example-cluster"
+      state: present
+      encrypted: true
+      master_username: 'dbadmin'
+      master_password: 'Test12345'
+      retention_period: 7
+      skip_final_snapshot: true
+'''
+
+RETURN = r'''
+db_cluster_arn:
+    description: The ARN of the graph database cluster you just created or updated.
+    returned: always
+    type: string
+'''
+
+import time
+
+try:
+    import botocore
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, AWSRetry
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, boto3_tag_list_to_ansible_dict
+
+
+def cluster_exists(client, module, params, result):
+    try:
+        response = client.describe_db_clusters(
+            DBClusterIdentifier=params['DBClusterIdentifier']
+        )
+        result['current_config'] = response['DBClusters'][0]
+        result['db_cluster_arn'] = response['DBClusters'][0]['DBClusterArn']
+        return True
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError):
```

Comment:
Returning False for any of these exceptions can mask important errors. Can you just catch the one specific to the DB cluster not existing to return False and other exceptions call fail_json_aws? Here's an example from another module for how to handle a specific exception code differently:
https://github.com/ansible/ansible/blob/42368ce3101309a51c0f29320d296328c28df918/lib/ansible/modules/cloud/amazon/rds_instance_facts.py#L366-L369